Framework/DataHandling/test/SaveNXSPETest.h

```diff
@@ -86,11 +127,64 @@ class SaveNXSPETest : public CxxTest::TestSuite {
     TS_ASSERT_DELTA(58.0, error[dims[0] * dims[1] - 1], tolerance);
   }
 
+  void test_Saving_Workspace_Small_EiOnLog_direct() {
+    // Create a small test workspace
+    const int nhist(3), nx(10);
+    MatrixWorkspace_sptr input = makeWorkspace(nhist, nx);
+    std::unique_ptr<Mantid::Kernel::PropertyWithValue<std::string>> mode(
+        new Mantid::Kernel::PropertyWithValue<std::string>("deltaE-mode", "Direct"));
+    // pointer is owned by run, no need to delete
+    input->mutableRun().addLogData(mode.release());
+
+    std::unique_ptr<Mantid::Kernel::PropertyWithValue<double>> EiLog(
+        new Mantid::Kernel::PropertyWithValue<double>("Ei", 12.0));
+    input->mutableRun().addLogData(EiLog.release());
```

Comment:
It looks like similar setup code is used in a few of the tests. Would it be possible to move this common setup into a separate re-usable function?